CONTRIBUTING.md

```diff
@@ -0,0 +1,13 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
```

Comment:
FWIW as an open source maintainer, I find I get a fair number of PRs out of the blue that are actually pretty useful.  You could still call this out as an open option, though - "feel free to solicit feedback" or whatever.

This is 100% up to you, though.

Reply:
Yeah I was one the fence there... I think I'll drop that line.  We can always discuss it in the PR.